weave/trace/op.py

```diff
@@ -535,6 +633,428 @@ def on_output(output: Any) -> Any:
     return res, call
 
 
+def _call_sync_gen(
+    op: Op,
+    *args: Any,
+    __weave: WeaveKwargs | None = None,
+    __should_raise: bool = False,
+    **kwargs: Any,
+) -> tuple[Generator[Any, None, None], Call]:
+    func = op.resolve_fn
+    call = placeholder_call()
+
+    # Handle all of the possible cases where we would skip tracing.
+    if should_skip_tracing_for_op(op):
+        gen = func(*args, **kwargs)
+        call.output = gen
+        return gen, call
+
+    if _should_sample_traces(op):
```

Comment:
I don't really understand this block. If `_should_sample_traces` is True, why do we disable tracing?

Reply:
Good catch, there should be a sampling param here that is missing

Reply:
I guess it reads better as `if not _should_sample_traces(op)` -- then that function should be inverted.

The intent here is to say "if we are sampling, then dont always log everything"